x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/support/SamlInitiateSingleSignOnAttributes.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.idp.saml.support;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ValidateActions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a collection of SAML attributes to be included in the SAML response.
+ * Each attribute has a key and a list of values.
+ */
+public class SamlInitiateSingleSignOnAttributes implements Writeable, ToXContentObject {
+    private Map<String, List<String>> attributes;
+
+    public SamlInitiateSingleSignOnAttributes() {
+        this.attributes = new HashMap<>();
+    }
+
+    /**
+     * @return A map of SAML attribute key to list of values
+     */
+    public Map<String, List<String>> getAttributes() {
+        return Collections.unmodifiableMap(attributes);
+    }
+
+    public SamlInitiateSingleSignOnAttributes setAttributes(Map<String, List<String>> attributes) {
+        this.attributes = new HashMap<>(attributes);
+        return this;
+    }
+
+    /**
+     * Creates a SamlInitiateSingleSignOnAttributes object by parsing the provided JSON content.
+     * Expects a JSON structure like: { "attr1": ["val1", "val2"], "attr2": ["val3"] }
+     *
+     * @param parser The XContentParser positioned at the start of the object
+     * @return A new SamlInitiateSingleSignOnAttributes instance
+     */
+    public static SamlInitiateSingleSignOnAttributes fromXContent(XContentParser parser) throws IOException {
+        Map<String, List<String>> attributes = new HashMap<>();
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String key = parser.currentName();
+            if (parser.nextToken() == XContentParser.Token.START_ARRAY) {
+                List<String> values = new ArrayList<>();
+                while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                    values.add(parser.text());
+                }
+                attributes.put(key, values);
+            }
+        }
+        SamlInitiateSingleSignOnAttributes attributesObj = new SamlInitiateSingleSignOnAttributes();
+        attributesObj.setAttributes(attributes);
+        return attributesObj;
```

Comment:
You should be able to replace all of this with:
```
        final Map<String, List<String>> attributes = parser.map(HashMap::new, p -> XContentParserUtils.parseList(p, XContentParser::text));
        return new SamlInitiateSingleSignOnAttributes(attributes);
```

Reply:
Done thanks.